src/v0/destinations/salesforce/utils.js

```diff
@@ -27,7 +27,11 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey) => {
   const { status, response } = destResponse;
 
   // if the response from destination is not a success case build an explicit error
-  if (!isHttpStatusSuccess(status) && status >= 400) {
+  if (
+    // This condition will pop-up when `invalid_grant` error is thrown from Salesforce
+    !Number.isInteger(status) ||
```

Comment:
@sanpj2292 If status is not an integer, the condition returns true & the block executes. Can you re-check whether that's the intended behaviour ?

Reply:
Raised a revert PR for this fix #1780

Reply:
Resolving this, as the revert has already taken place